nerfstudio/scripts/render.py

```diff
@@ -129,6 +140,46 @@ def _render_trajectory_video(
                     bounding_box_max = crop_data.center + crop_data.scale / 2.0
                     aabb_box = SceneBox(torch.stack([bounding_box_min, bounding_box_max]).to(pipeline.device))
                 camera_ray_bundle = cameras.generate_rays(camera_indices=camera_idx, aabb_box=aabb_box)
+                if "nearest_camera" in rendered_output_names:
+                    cam_pos = cameras[camera_idx].camera_to_worlds[:, 3].cpu()
+                    cam_rot = R.from_matrix(cameras[camera_idx].camera_to_worlds[:3, :3].cpu())
+                    cam_quat = cam_rot.as_quat()
+
+                    max_dist, max_idx = None, None
+                    true_max_dist, true_max_idx = None, None
+                    for i in range(len(train_cameras)):
+                        train_cam_pos = train_cameras[i].camera_to_worlds[:, 3].cpu()
+                        # Make sure the line of sight from rendered cam to training cam is not blocked by any object
+                        bundle = RayBundle(
+                            origins=cam_pos.view(1, 3),
+                            directions=((cam_pos - train_cam_pos) / (cam_pos - train_cam_pos).norm()).view(1, 3),
+                            pixel_area=torch.tensor(1).view(1, 1),
+                            nears=torch.tensor(0.05).view(1, 1),
+                            fars=torch.tensor(100).view(1, 1),
+                            camera_indices=torch.tensor(0).view(1, 1),
+                            metadata={},
+                        ).to(pipeline.device)
+                        outputs = pipeline.model.get_outputs(bundle)
+
+                        r = R.from_matrix(train_cameras[i].camera_to_worlds[:3, :3].cpu())
+                        q = r.as_quat()
+                        # calculate distance between two quaternions
+                        rot_dist = 1 - np.dot(q, cam_quat) ** 2
+                        pos_dist = torch.norm(train_cam_pos - cam_pos)
+                        dist = 0.3 * rot_dist + 0.7 * pos_dist
+
+                        if outputs["depth"][0] < torch.norm(cam_pos - train_cam_pos).item():
+                            if true_max_dist is None or dist < true_max_dist:
+                                true_max_dist = dist
+                                true_max_idx = i
+                            continue
+
+                        if max_dist is None or dist < max_dist:
+                            max_dist = dist
+                            max_idx = i
+
+                    if max_idx is None:
```

Comment:
I think this will never be done? Because of the lines above `if max_dist is None or dist < max_dist:`

Reply:
Actually this will occur in cases where every camera is occluded because of the continue in the if statement above